src/types/ButtonProps.ts

```diff
@@ -1,8 +1,8 @@
 export type ButtonProps = {
-    title: string;
+    title?: string;
     onClick?: () => void | Promise<void>; // optional, but recommended to handle async operations
-    //For the places where we will use green buttons
+    //To specify which color the button will be
     color: string;
-    //FIXME: add correct typing
-    icon?: any;
+    cssClass?: string;
+    icon?: string;
```

Comment:
What kind of strong is this? Based on how it's used in the code it seems it should be valid HTML? I'd say at least add a comment here saying what kind of string it should, but even better would be to emulate a _nominal type_, ie. a unique type which you explicitly assign to a variable via a type casting/checking function:


```suggestion
    // Define your nominal type
export declare const HtmlStringBrand: unique symbol;
export type HtmlString = string & { [HtmlStringBrand]: true };

// Create a function to cast a string to your nominal type
export function createHtmlString(html: string): HtmlString {
    /*
      Add some checks here
    */

    return html as HtmlString;
}

```

Reply:
Yes, I understand what you mean!. However I have changed the PNG and we won't need to pass strings as icons. Instead, we will use the icons like this:
( Create React App (CRA),  comes with built-in support for SVG imports as React components)
`
/*
in ButtonProps.ts :

*/
 icon?: ReactComponentElement<any, any>; // This type allows us to pass a JSX element (e.g., <SomeSvg />) as the icon prop.

/*
Login.tsx example

/*
We can import the svg directly from the file as a react icon
*/
import { ReactComponent as CheckIcon } from "../../assets/icons/check-icon.svg";
//The ReactComponent import syntax ensures that the SVG content is encapsulated correctly as a React component

{....}

<Button
	color="pink"
	icon={<CheckIcon/>}
	/>
*/
`

I hope this is clear enough otherwise let me know and I can explain further! I have already added the SVG icons but they'r ein the other PR! @plundell